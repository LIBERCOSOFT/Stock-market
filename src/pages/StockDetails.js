import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';

import { fetchStockProfile } from '../redux/details/details';

const StockDetails = () => {
  const stockProfile = useSelector((state) => state.stockProfile);

  const { symbol } = useParams();
  const dispatch = useDispatch();

  useEffect(() => {
    if (stockProfile.length < 1 || stockProfile[0].symbol !== symbol) {
      dispatch(fetchStockProfile(symbol));
    }
  }, [dispatch, stockProfile, symbol]);

  return (
    <div className="details__container">
      <img src={stockProfile[0].image} alt="" />
      <p>
        <span>DESCRIPTION:</span>
        {stockProfile[0].description}
      </p>
      <div>

        <p>
          <span>NAME:</span>
          {stockProfile[0].companyName}
        </p>
        <p>
          <span>CEO:</span>
          {' '}
          {stockProfile[0].ceo}
        </p>
        <p>
          <span>WEBSITE:</span>
          {' '}
          {stockProfile[0].website}
        </p>
        <p>
          <span>INDUSTRY:</span>
          {' '}
          {stockProfile[0].industry}
        </p>
        <p>
          <span>Address:</span>
          {' '}
          {stockProfile[0].address}
          ,
          {' '}
          {stockProfile[0].city}
          ,
          {' '}
          {stockProfile[0].state}
          ,
          {' '}
          {stockProfile[0].country}
          .
        </p>
        <p>
          <span>SYMBOL:</span>
          {' '}
          {stockProfile[0].symbol}
        </p>
        <p>
          <span>Price Per Stock:</span>
          {' '}
          $
          {stockProfile[0].price || 0}
        </p>
        <p>
          <span>Market Capital:</span>
          {' '}
          $
          {(stockProfile[0].mktCap || 0).toLocaleString()}
        </p>
      </div>
    </div>
  );
};

export default StockDetails;
