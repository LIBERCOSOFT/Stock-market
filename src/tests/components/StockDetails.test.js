import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import StockDetails from '../../pages/StockDetails';

test('renders stock-details component snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <StockDetails />
      </BrowserRouter>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
