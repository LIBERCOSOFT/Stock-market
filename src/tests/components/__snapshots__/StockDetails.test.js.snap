// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders stock-details component snapshot 1`] = `
<DocumentFragment>
  <div
    class="details__container"
  >
    <img
      alt=""
    />
    <p>
      <span>
        DESCRIPTION:
      </span>
    </p>
    <div>
      <p>
        <span>
          NAME:
        </span>
      </p>
      <p>
        <span>
          CEO:
        </span>
         
      </p>
      <p>
        <span>
          WEBSITE:
        </span>
         
      </p>
      <p>
        <span>
          INDUSTRY:
        </span>
         
      </p>
      <p>
        <span>
          Address:
        </span>
         , , , .
      </p>
      <p>
        <span>
          SYMBOL:
        </span>
         
      </p>
      <p>
        <span>
          Price Per Stock:
        </span>
         $0
      </p>
      <p>
        <span>
          Market Capital:
        </span>
         $0
      </p>
    </div>
  </div>
</DocumentFragment>
`;
